ethereum/verkletrie/src/main/java/org/hyperledger/besu/ethereum/verkletrie/bandersnatch/fp/Element.java

```diff
@@ -0,0 +1,522 @@
+/*
+ * Copyright Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.verkletrie.bandersnatch.fp;
+
+import java.math.BigInteger;
+import java.nio.ByteOrder;
+import java.util.Objects;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.apache.tuweni.bytes.MutableBytes32;
+import org.apache.tuweni.units.bigints.UInt256;
+
+public class Element {
+  public static final Element ZERO = new Element(UInt256.ZERO);
+  public static final Element ONE;
+  static final Element Q_MODULUS;
+  private static final Element R_SQUARE;
+
+  static {
+    {
+      UInt256 z0 = UInt256.valueOf(8589934590L);
```

Comment:
I think it would be nice to have some info. Know what these values ​​are related to for the next developers who will look at this part

Reply:
Sorry, not sure myself. I got them from the original project.